### PR TITLE
Add validation for the metrics data

### LINKED ISSUE #1000: Handling missing metrics[] with right messaging
If Cost operator does not send metrics[] appropriate messaging should be handled

### PR DESCRIPTION
**Pull request by khansaad**
This PR fixes #1000

### CODE REVIEW

**Comment by dinogun** [comment]
@msvinaykumar @chandrams Please provide your reviews

File: src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java [hunk 1]
```diff
@@ -191,7 +191,10 @@ public void validateAndAddExperimentResults(List<UpdateResultsAPIObject> updateR
         List<ExperimentResultData> resultDataList = new ArrayList<>();
         successUpdateResultsAPIObjects.forEach(
                 (successObj) -> {
-                    resultDataList.add(Converters.KruizeObjectConverters.convertUpdateResultsAPIObjToExperimentResultData(successObj));
+                    ExperimentResultData experimentResultData = Converters.KruizeObjectConverters.convertUpdateResultsAPIObjToExperimentResultData(successObj);
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java:194]: experimentResultData.getValidationOutputData().isSuccess() 
is not success then how we are going to report to user 
is it getting added to 
failedUpdateResultsAPIObjects  ?

Comment (msvinaykumar) [reply src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java:194]: To test this secanrio try throw Exception manually and see if it is reported back to response

File: src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java [hunk 2]
```diff
@@ -85,6 +85,7 @@ public static final class AutotuneObjectErrors {
         public static final String WRONG_TIMESTAMP = "The Start time should precede the End time!";
 
         public static final String MEASUREMENT_DURATION_ERROR = "Interval duration cannot be less than or greater than measurement_duration by more than " + KruizeConstants.TimeConv.MEASUREMENT_DURATION_THRESHOLD_SECONDS + " seconds";
+        public static final String MISSING_METRICS = "Metric data is not present for container : %s for experiment: %s";
```
Comment (chandrams) [inline src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java:88]: @khansaad - We do not have tests where complete metrics section is missing at present, these need to be added and validated against the expected response message.

Comment (chandrams) [reply src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java:88]: Also run the existing negative test that pass null for these metrics and check if the expected messages manually.

File: src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java (hunk 1 above)
Comment (khansaad) [reply src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java:194]: I tried adding multiple objects and failed a couple of them to check the response. Here's what I'm getting:

```
{
    "message": "Out of a total of 3 records, 2 failed to save",
    "httpcode": 400,
    "documentationLink": "",
    "status": "ERROR",
    "data": [
        {
            "interval_start_time": "2023-09-10T14:00:00.000Z",
            "interval_end_time": "2023-09-10T14:15:00.000Z",
            "errors": [
                {
                    "message": "kubernetes_objects : Metric data is not present for container : tfb-server-0 for experiment: 5kexp_1",
                    "httpcode": 400,
                    "documentationLink": "",
                    "status": "ERROR"
                }
            ],
            "version": "1.0",
            "experiment_name": "5kexp_1"
        },
        {
            "interval_start_time": "2023-09-10T15:30:00.000Z",
            "interval_end_time": "2023-09-10T15:45:00.000Z",
            "errors": [
                {
                    "message": "kubernetes_objects : Metric data is not present for container : tfb-server-0 for experiment: 5kexp_1",
                    "httpcode": 400,
                    "documentationLink": "",
                    "status": "ERROR"
                }
            ],
            "version": "1.0",
            "experiment_name": "5kexp_1"
        }
    ]
}
```

**Comment by chandrams** [comment]
@khansaad @msvinaykumar I have included a few tests with missing metrics here #1009 please use these and update other jsons that you are testing against to it.

**Review by msvinaykumar — APPROVED** [review]
LGTM

**Review by chandrams — APPROVED** [review]
Tested on minikube, changes LGTM!
